From 572494547b6ab34fc624f9efda0b3b28fcf4a4cd Mon Sep 17 00:00:00 2001
From: datauwu <209150516+datauwu@users.noreply.github.com>
Date: Wed, 14 Oct 2026 23:14:40 +0800
Subject: [PATCH] autotools: fix libxml2 linking

The custom AC_CHECK_LIB success action prevents HAVE_LIBXML2 from
being defined, even when libxml2 is detected.

Explicitly define HAVE_LIBXML2 on success to restore libxml2
support and linking.

Origin: upstream, https://github.com/libarchive/libarchive/commit/572494547b6ab34fc624f9efda0b3b28fcf4a4cd
Forwarded: not-needed
Last-Update: 2026-10-15
---
 configure.ac | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 02ca370bcf..b00eed1367 100644
--- a/configure.ac
+++ b/configure.ac
@@ -585,7 +585,9 @@ if test "x$with_xml2" != "xno"; then
   PKG_CHECK_MODULES(LIBXML2_PC, [libxml-2.0 >= 2.7.3], [
     CPPFLAGS="${CPPFLAGS} ${LIBXML2_PC_CFLAGS}"
     LIBS="${LIBS} ${LIBXML2_PC_LIBS}"
-    AC_CHECK_LIB(xml2,xmlInitParser,[true],AC_MSG_FAILURE(Missing xml2 library))
+    AC_CHECK_LIB(xml2,xmlInitParser,
+      [AC_DEFINE([HAVE_LIBXML2], [1], [Define to 1 if you have the 'xml2' library.])],
+      [AC_MSG_FAILURE([Missing xml2 library])])
   ], [
     AC_CHECK_LIB(xml2,xmlInitParser)
   ])
-- 
2.51.0

